### ISSUE TITLE
Heroku build fails

### ISSUE THREAD

**Issue by peterzoller** [vapor-community/heroku-buildpack#30]
Hey! 
Do you have any idea, why my build fails? It fails on different libs any time i'd like to deploy it to heroku.  The error log does not say anything useful.
Thanks!
Here is the log:

```
Counting objects: 1241, done.
Delta compression using up to 8 threads.
Compressing objects: 100% (882/882), done.
Writing objects: 100% (1241/1241), 911.43 KiB | 4.34 MiB/s, done.
Total 1241 (delta 572), reused 468 (delta 209)
remote: Compressing source files... done.
remote: Building source:
remote: 
remote: -----> Swift app detected
remote: -----> Installing clang 5.0.0
remote: -----> Installing swiftenv
remote: Cloning into 'swiftenv'...
remote: -----> Installing Swift 4.1.2
remote: Downloading https://swift.org/builds/swift-4.1.2-release/ubuntu1604/swift-4.1.2-RELEASE/swift-4.1.2-RELEASE-ubuntu16.04.tar.gz
remote: /tmp/swiftenv-4.1.2- ~/tmp/buildpacks/2db5e06b2426c8778ade338f27d60743ddf42f45cb2e38e7cf3b2511145bc4400ce84a93ddec59b96a39485d2c2394c4c489c7fc98a2d0faa3515f3c90da67a9
remote:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
remote:                                  Dload  Upload   Total   Spent    Left  Speed
remote: 100  171M  100  171M    0     0  8510k      0  0:00:20  0:00:20 --:--:-- 9161k
remote: ~/tmp/buildpacks/2db5e06b2426c8778ade338f27d60743ddf42f45cb2e38e7cf3b2511145bc4400ce84a93ddec59b96a39485d2c2394c4c489c7fc98a2d0faa3515f3c90da67a9
remote: 4.1.2 has been installed.
remote: -----> Building package (release configuration)
remote: Compile CNIOAtomics src/c-atomics.c
remote: Compile CNIOSHA1 c_nio_sha1.c
remote: Compile CNIOLinux shim.c
remote: Compile CNIOLinux ifaddrs-android.c
remote: Compile CNIODarwin shim.c
remote: Compile CNIOZlib empty.c
remote: Compile CNIOHTTPParser c_nio_http_parser.c
remote: Compile CNIOOpenSSL shims.c
remote: Compile CNIOOpenSSL helpers.c
remote: Compile CCryptoOpenSSL shim.c
remote: Compile CBcrypt blf.c
remote: Compile CBcrypt bcrypt.c
remote: Compile CBase32 base32.c
remote: Compile Swift Module 'NIOPriorityQueue' (2 sources)
remote: Compile Swift Module 'Debugging' (3 sources)
remote: Compile Swift Module 'COperatingSystem' (1 sources)
remote: Compile Swift Module 'NIOConcurrencyHelpers' (2 sources)
remote: Compile Swift Module 'NIO' (55 sources)
remote: Compile Swift Module 'NIOTLS' (3 sources)
remote: Compile Swift Module 'NIOHTTP1' (9 sources)
remote: Compile Swift Module 'NIOFoundationCompat' (1 sources)
remote: Compile Swift Module 'Bits' (12 sources)
remote: Compile Swift Module 'Async' (15 sources)
remote: Compile Swift Module 'NIOOpenSSL' (15 sources)
remote: Compile Swift Module 'Random' (4 sources)
remote: Compile Swift Module 'Core' (25 sources)
remote: /tmp/build_022abe05b7761ccd00d3bd45b0fbccb9/.build/checkouts/swift-nio-ssl.git-574086289262766763/Sources/NIOOpenSSL/OpenSSLHandler.swift:270:35: warning: 'changeCapacity(to:)' is deprecated: changeCapacity has been replaced by reserveCapacity
remote:                     receiveBuffer.changeCapacity(to: receiveBuffer.capacity + 1024)
remote:                                   ^
remote: Compile Swift Module 'NIOWebSocket' (9 sources)
remote: Compile Swift Module 'HTTP' (25 sources)
remote: Compile Swift Module 'Crypto' (19 sources)
remote: Compile Swift Module 'Validation' (18 sources)
remote: Compile Swift Module 'URLEncodedForm' (8 sources)
remote: Compile Swift Module 'Service' (20 sources)
remote: Compile Swift Module 'Multipart' (8 sources)
remote: Compile Swift Module 'Logging' (4 sources)
remote: Compile Swift Module 'WebSocket' (6 sources)
remote: Compile Swift Module 'TemplateKit' (41 sources)
remote: Compile Swift Module 'Routing' (12 sources)
remote: Compile Swift Module 'DatabaseKit' (30 sources)
remote: Compile Swift Module 'Console' (27 sources)
remote: Compile Swift Module 'SQL' (59 sources)
remote: Compile Swift Module 'Command' (16 sources)
remote: Compile Swift Module 'Vapor' (75 sources)
remote: Compile Swift Module 'Fluent' (49 sources)
remote: Compile Swift Module 'PostgreSQL' (72 sources)
remote: Compile Swift Module 'FluentSQL' (9 sources)
remote: Compile Swift Module 'VaporTestTools' (16 sources)
remote: /tmp/build_022abe05b7761ccd00d3bd45b0fbccb9/.build/checkouts/VaporTestTools.git--4928636270379599054/Sources/VaporTestTools/Extensions/Requests/HTTPRequest+Response.swift:10:29: error: module 'NIO' was not compiled for testing
remote: @_exported @testable import NIO
remote:                             ^
remote: /tmp/build_022abe05b7761ccd00d3bd45b0fbccb9/.build/checkouts/VaporTestTools.git--4928636270379599054/Sources/VaporTestTools/Extensions/Response tools/Response+Checks.swift:9:18: error: module 'Vapor' was not compiled for testing
remote: @testable import Vapor
remote:                  ^
remote: /tmp/build_022abe05b7761ccd00d3bd45b0fbccb9/.build/checkouts/VaporTestTools.git--4928636270379599054/Sources/VaporTestTools/Extensions/Response tools/Response+Checks.swift:10:18: error: module 'NIO' was not compiled for testing
remote: @testable import NIO
remote:                  ^
remote: /tmp/build_022abe05b7761ccd00d3bd45b0fbccb9/.build/checkouts/VaporTestTools.git--4928636270379599054/Sources/VaporTestTools/Extensions/Response tools/Response+Debug.swift:9:18: error: module 'Vapor' was not compiled for testing
remote: @testable import Vapor
remote:                  ^
remote: /tmp/build_022abe05b7761ccd00d3bd45b0fbccb9/.build/checkouts/VaporTestTools.git--4928636270379599054/Sources/VaporTestTools/Extensions/Response tools/Response+Getters.swift:9:18: error: module 'Vapor' was not compiled for testing
remote: @testable import Vapor
remote:                  ^
remote: /tmp/build_022abe05b7761ccd00d3bd45b0fbccb9/.build/checkouts/VaporTestTools.git--4928636270379599054/Sources/VaporTestTools/Extensions/Response tools/Response+Getters.swift:10:18: error: module 'NIO' was not compiled for testing
remote: @testable import NIO
remote:                  ^
remote: /tmp/build_022abe05b7761ccd00d3bd45b0fbccb9/.build/checkouts/VaporTestTools.git--4928636270379599054/Sources/VaporTestTools/Extensions/Services/Services+Testing.swift:9:18: error: module 'Service' was not compiled for testing
remote: @testable import Service
remote:                  ^
remote: error: terminated(1): /app/tmp/cache/heroku-16/swiftenv/versions/4.1.2/usr/bin/swift-build-tool -f /tmp/build_022abe05b7761ccd00d3bd45b0fbccb9/.build/release.yaml main output:
remote:     
remote: 
remote: Fetching https://github.com/vapor/multipart.git
remote: Fetching https://github.com/vapor/crypto.git
remote: Fetching https://github.com/vapor/http.git
remote: Fetching https://github.com/vapor/core.git
remote: Fetching https://github.com/vapor/postgresql.git
remote: Fetching https://github.com/apple/swift-nio.git
remote: Fetching https://github.com/vapor/database-kit.git
remote: Fetching https://github.com/vapor/service.git
remote: Fetching https://github.com/vapor/sql.git
remote: Fetching https://github.com/vapor/websocket.git
remote: Fetching https://github.com/vapor/vapor.git
remote: Fetching https://github.com/vapor/fluent-postgresql
remote: Fetching https://github.com/vapor/fluent.git
remote: Fetching https://github.com/vapor/url-encoded-form.git
remote: Fetching https://github.com/vapor/auth.git
remote: Fetching https://github.com/apple/swift-nio-zlib-support.git
remote: Fetching https://github.com/vapor/routing.git
remote: Fetching https://github.com/vapor/console.git
remote: Fetching https://github.com/apple/swift-nio-ssl.git
remote: Fetching https://github.com/vapor/template-kit.git
remote: Fetching https://github.com/twof/VaporMailgunService.git
remote: Fetching https://github.com/apple/swift-nio-ssl-support.git
remote: Fetching https://github.com/vapor/validation.git
remote: Fetching https://github.com/LiveUI/VaporTestTools.git
remote: Cloning https://github.com/apple/swift-nio.git
remote: Resolving https://github.com/apple/swift-nio.git at 1.10.0
remote: Cloning https://github.com/vapor/core.git
remote: Resolving https://github.com/vapor/core.git at 3.4.4
remote: Cloning https://github.com/vapor/sql.git
remote: Resolving https://github.com/vapor/sql.git at 2.2.0
remote: Cloning https://github.com/vapor/websocket.git
remote: Resolving https://github.com/vapor/websocket.git at 1.1.0
remote: Cloning https://github.com/vapor/fluent-postgresql
remote: Resolving https://github.com/vapor/fluent-postgresql at 1.0.0
remote: Cloning https://github.com/vapor/url-encoded-form.git
remote: Resolving https://github.com/vapor/url-encoded-form.git at 1.0.5
remote: Cloning https://github.com/vapor/auth.git
remote: Resolving https://github.com/vapor/auth.git at 2.0.1
remote: Cloning https://github.com/apple/swift-nio-zlib-support.git
remote: Resolving https://github.com/apple/swift-nio-zlib-support.git at 1.0.0
remote: Cloning https://github.com/apple/swift-nio-ssl.git
remote: Resolving https://github.com/apple/swift-nio-ssl.git at 1.3.2
remote: Cloning https://github.com/vapor/template-kit.git
remote: Resolving https://github.com/vapor/template-kit.git at 1.1.0
remote: Cloning https://github.com/LiveUI/VaporTestTools.git
remote: Resolving https://github.com/LiveUI/VaporTestTools.git at master
remote: Cloning https://github.com/apple/swift-nio-ssl-support.git
remote: Resolving https://github.com/apple/swift-nio-ssl-support.git at 1.0.0
remote: Cloning https://github.com/vapor/multipart.git
remote: Resolving https://github.com/vapor/multipart.git at 3.0.2
remote: Cloning https://github.com/vapor/crypto.git
remote: Resolving https://github.com/vapor/crypto.git at 3.3.0
remote: Cloning https://github.com/vapor/http.git
remote: Resolving https://github.com/vapor/http.git at 3.1.6
remote: Cloning https://github.com/vapor/postgresql.git
remote: Resolving https://github.com/vapor/postgresql.git at 1.1.0
remote: Cloning https://github.com/vapor/database-kit.git
remote: Resolving https://github.com/vapor/database-kit.git at 1.3.1
remote: Cloning https://github.com/vapor/service.git
remote: Resolving https://github.com/vapor/service.git at 1.0.0
remote: Cloning https://github.com/vapor/vapor.git
remote: Resolving https://github.com/vapor/vapor.git at 3.1.0
remote: Cloning https://github.com/vapor/fluent.git
remote: Resolving https://github.com/vapor/fluent.git at 3.1.2
remote: Cloning https://github.com/vapor/routing.git
remote: Resolving https://github.com/vapor/routing.git at 3.0.1
remote: Cloning https://github.com/vapor/console.git
remote: Resolving https://github.com/vapor/console.git at 3.0.3
remote: Cloning https://github.com/twof/VaporMailgunService.git
remote: Resolving https://github.com/twof/VaporMailgunService.git at 1.5.0
remote: Cloning https://github.com/vapor/validation.git
remote: Resolving https://github.com/vapor/validation.git at 2.1.1
remote:  !     Push rejected, failed to compile Swift app.
remote: 
remote:  !     Push failed
remote: Verifying deploy...
remote: 
remote: !	Push rejected to .....
remote: 
To https://git.heroku.com/......git
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'https://git.heroku.com.....git'

```

**Comment by RuiAAPeres** [comment]
Any idea why this was happening? I am stuck on this as well.

**Comment by vzsg** [comment]
With the exact same build output?

**Comment by RuiAAPeres** [comment]
There was something from my side, unrelated to vapor. In my particular case, `arc4random ` wasn't available in Linux. The message was buried and I could only see the latest bits:

```
remote: Cloning https://github.com/vapor/validation.git
remote: Resolving https://github.com/vapor/validation.git at 2.1.1
remote:  !     Push rejected, failed to compile Swift app.
```

**Comment by vzsg** [comment]
Okay, that's a completely different issue then. The `arc4random` function indeed isn't available, but since Swift 4.2, generating random numbers and shuffling arrays is now possible with the standard library.
